src/pymorize/rule.py

```diff
@@ -12,6 +13,51 @@
 from .data_request.variable import DataRequestVariable
 from .gather_inputs import InputFileCollection
 from .logging import logger
+from .pipeline import Pipeline
+
+
+class RuleRequirement:
+    """Defines a requirement which steps must have tagged in order to be valid"""
+
+    def __init__(self, requirement_name: str, requirement_value: typing.Any):
+        """
+        Initialize a RuleRequirement object.
+
+        Parameters
+        ----------
+        requirement_name : str
+            The name of the requirement that the step must satisfy.
+        requirement_value : Any
+            The value of the requirement that the step must satisfy.
+        """
+        self.requirement_name = requirement_name
+        self.requirement_value = requirement_value
+
+    @classmethod
+    def from_dict(cls, d):
+        """Build a RuleRequirement object from a dictionary"""
+        return cls(**d)
+
+    def pipeline_fulfills_requirements(self, pipeline: Pipeline) -> bool:
+        """Check if a pipeline fulfills the requirements of the rule
+
+        Parameters
+        ----------
+        pipeline : Pipeline
+            The pipeline to check
+
+        Returns
+        -------
+        bool
+            True if the pipeline fulfills the requirements, False otherwise
+        """
+        for step in pipeline.steps:
+            step_src = inspect.getsource(step)
+            for line in step_src.split("\n"):
+                if f"_satisfies_{self.requirement_name}" in line:
+                    if str(self.requirement_value) in line:
```

Comment:
I'm not terribly convinced by this, it requires that the name and value are strictly defined to be on the same line. That might not be the best way of doing it.